michbl/assignment1/assignment1.cpp

```diff
@@ -7,84 +7,129 @@ using namespace std;
 
 //Question 1 - Anagrams
 
-//check whether a character is an english letter
+// Assumes that given words are without spaces
+bool areWordsAnagrams(string word1, string word2, bool caseSensitive);
+
+bool isUpperCase(char c) {
+    return (c >= 'A' && c <= 'Z');
+}
+
+bool isLowerCase(char c) {
+    return (c >= 'a' && c <= 'z');
+}
+
+// Check whether a character is an english letter
 bool isLetter(char c) {
-    return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
+    return isUpperCase(c) || isLowerCase(c);
+}
+
+char toUpperCase(char c) {
+    if (!isLetter(c)) {
+        return 0;

```

Comment:
This is a little unexpected... Why turn a non-letter to zero?
If this is an 'error', I suggest you do the following:
1. Add a comment on this function which clarifies this, e.g.
`// Returns zero if input character is not a letter`
(because the function name does not reflect that).
2. Whenever you call this function and you did not precheck that the input is a letter, check that its output is not zero (otherwise this function is not safe to use). I see that in your code there are no such occurances.

Another option is to make it return the same character in case input is not a letter. I think it makes this function easier to use (for example, you can turn an entire sentence to uppercase by calling this function on each character, without checking for spaces and punctuation).